meteomatics/src/util.rs

```diff
@@ -0,0 +1,399 @@
+            format!(
+                "{}?{}",
+                query_specs,
+                optionals.as_ref().unwrap().join("&")
+            )
+        }
+    };
+    
+    query_specs
+}
+
+/// Builds the query specifications ('specs') for a time series grid query according to the Meteomatics 
+/// API format rules. Optionally parses a number of provided extra specifiers (e.g. 'model=mix'). The
+/// dates are formatted according to ISO8601 (<https://en.wikipedia.org/wiki/ISO_8601>). The format
+/// parameter specifies the requested file type (e.g. "csv" or "netcdf" or "png").
+pub async fn build_grid_ts_query_specs(
+    start_date: &chrono::DateTime<chrono::Utc>,
+    end_date: &chrono::DateTime<chrono::Utc>,
+    interval: &chrono::Duration,
+    parameter: &String,
+    coords_str: &str,
+    format: &str,
+    optionals: &Option<Vec<String>>,
+) -> String {
+    let query_specs = format!(
+        "{}--{}:{}/{}/{}/{}",
+        start_date.to_rfc3339(),
+        end_date.to_rfc3339(),
+        interval,
+        parameter,
+        coords_str,
+        format,
+    );
+
+    // Handles optional parameters 
+    let query_specs = match optionals {
+        None => query_specs,
+        Some(_) => {
+            format!(
+                "{}?{}",
+                query_specs,
+                optionals.as_ref().unwrap().join("&")
+            )
+        }
+    };
+    
+    query_specs
+}
+
+
+/// Combines the default base API URL with the query specific information.
+pub async fn build_url(url_fragment: &str) -> std::result::Result<Url, ParseError> {
+    let base_url = Url::parse(BASE_URL).expect("Base URL is known to be valid");
+    let full_url = base_url.join(url_fragment)?;
+    Ok(full_url)
+}
+
+/// Convert a number of Points to a String according to the Meteomatics API specifications.
+pub async fn points_to_str(coords: &[Point]) -> String {
+    coords.iter().map(|p| format!("{}", p)).collect::<Vec<String>>().join("+")
+}
+
+#[cfg(test)]
+mod tests {
+
+    use chrono::prelude::*;
+    use chrono::Duration;
+    use crate::location::{Point, BBox};
+    use std::path::Path;
+    use std::fs;
+
+
+    #[tokio::test]
+    async fn check_path_creation_nonexistent() {
+        let file_name: String = String::from("tests/netcdfs/my_netcdf.nc");
+        crate::util::create_path(&file_name).await.unwrap();
+        let dir: &Path = Path::new(&file_name).parent().unwrap();
+        let check: bool = dir.is_dir();
+        assert_eq!(check, true);
+        fs::remove_dir_all(dir).unwrap();
+    }
+
+    #[tokio::test]
+    // checks if the location specifier is correctly created
+    async fn check_locations_string() {
+        let p1: Point = Point { lat: 52.520551, lon: 13.461804};
+        let p2: Point = Point { lat: -52.520551, lon: 13.461804};
+        let coords: Vec<Point> = vec![p1, p2];
+        let coord_str = crate::util::points_to_str(&coords).await;
+        assert_eq!("52.520551,13.461804+-52.520551,13.461804", coord_str);
```

Comment:
2 magic variables, I would name them at the top and use them for Point struct and assert, otherwise if they change you will have to change them in 2 different places, it is prone to bugs then.